.github/workflows/darwin-tests.yaml

```diff
@@ -38,7 +38,17 @@ jobs:
 
     strategy:
       matrix:
-        build_variant: [no-ble-no-shell-asan-clang]
+        include:
+        - os: macos-14
+          arch: arm64     # observed, not configured
+          dry-run: true
+          build_variant: no-ble-no-shell-asan-clang
+          chip_tool: ""
+        - os: macos-15
```

Comment:
Does this add even more CI to darwin? darwin is already the main critical path/slow runner we have.

Reply:
At least could we just run the extra bits on master only?

I am unclear if there is really an expectation that a macos15 will find more coverage than macos14 - I imagine for 99% or more the logic will be identical. Could we divide it up in things that we truly believe may be different?